cli/upload.py

```diff
@@ -265,29 +298,23 @@ def _gsutil_assay_upload(upload_info: api.UploadInfo, xlsx: str):
         except StopIteration:
             all_uploads_have_run = True
 
-        err = _wait_for_upload(procs, file_count)
-
-        if err:
-            for p in procs:
-
-                if p.poll() != 0:
-
-                    # stopping all other processes
-                    for other_p in procs:
-                        if p != other_p:
-                            other_p.kill()
-
-                    click.echo(
-                        f"\nGCS upload failed on {p.args[-2]} with the following message:\n"
-                    )
-                    click.secho(err, fg="red")
-
-                    raise click.Abort()
+        skipped = _wait_for_upload(
+            procs, file_count, upload_info.optional_files, skipped
+        )
 
     click.echo(
         f"[{file_count}/{file_count} done] All files uploaded to GCS and staged for ingestion."
     )
 
+    # remove the skipped files from the map
+    gcs_file_map = upload_info.gcs_file_map
+    for s in skipped:
+        print(s)
+        # turn skipped from indexes to gcs_uri
+        gcs_file_map.pop(procs[s].args[-1], {})
+
+    return gcs_file_map
+
 
 def _compose_file_mapping(upload_info: api.UploadInfo, xlsx: str):
```

Comment:
I think this is where we should be checking for missing optional files, since it will allow us to prune absent optional files from our GCS upload, rather than waiting for GCS uploads to fail and handling those errors in `wait_for_upload`. What do you think?

Reply:
in next commit